safedelete/utils.py

```diff
@@ -1,17 +1,24 @@
-from django.db import router
-from django.db.models.deletion import Collector
+import itertools
 
-HARD_DELETE = 0
-SOFT_DELETE = 1
-SOFT_DELETE_CASCADE = 2
-HARD_DELETE_NOCASCADE = 3
-NO_DELETE = 4
+from django.db import router
 
-DELETED_INVISIBLE = 10
-DELETED_VISIBLE_BY_PK = 11
+try:
+    # Django 1.7
+    from django.contrib.admin.utils import NestedObjects
+except ImportError:
+    # Django < 1.7
+    from django.contrib.admin.util import NestedObjects
 
 
 def can_hard_delete(obj):
-    collector = Collector(using=router.db_for_write(obj))
+    collector = NestedObjects(using=router.db_for_write(obj))
     collector.collect([obj])
-    return not bool(collector.dependencies)
+
+    def flatten(elem):
+        if isinstance(elem, list):
+            return itertools.chain.from_iterable(map(flatten, elem))
+        elif obj != elem:
+            return (elem,)
+        return ()
+
+    return not bool(list(flatten(collector.nested())))
```

Comment:
I have reverted the change made here that moved to directly using the Collector. Neither `Collector.fast_deletes` nor `Collector.dependencies` were given the models that were gonna be deleted by cascading and your previous implementation seems to do.